lib/matplotlib/tests/test_legend.py

```diff
@@ -1450,3 +1450,38 @@ def test_boxplot_labels():
     handles, labels = ax.get_legend_handles_labels()
     assert len(handles) == 0
     assert len(labels) == 0
+
+
+def test_boxplot_legend_labels():
+    # Test that legend entries are generated when passing `label`.
+    np.random.seed(19680801)
+    data = np.random.random((10, 4))
+    fig, axs = plt.subplots(nrows=1, ncols=4)
+    legend_labels = ['A', 'B', 'C', 'D']
+
+    # Testing legend labels and patch passed to legend.
+    bp1 = axs[0].boxplot(data, patch_artist=True, label=legend_labels)
+    for i, v in enumerate(bp1['boxes']):
+        assert v.get_label() == legend_labels[i]
```

Comment:
Optional, but IMHO more concise:
```suggestion
    assert [v.get_label() for v in bp1['boxes']] == legend_labels 
```